napari/layers/volume/volume.py

```diff
@@ -0,0 +1,224 @@
+        dynamically loaded.
+    name : str, keyword-only
+        Name of the layer.
+
+    Attributes
+    ----------
+    data : array
+        Volume data. Can be 3 dimensional. If the last dimenstion (channel) is 3
+        or 4 can be interpreted as RGB or RGBA if multichannel is `True`.
+    metadata : dict
+        Volume metadata.
+    multichannel : bool
+        Whether the volume is multichannel RGB or RGBA if multichannel. If not
+        specified by user and the last dimension of the data has length 3 or 4
+        it will be set as `True`. If `False` the volume is interpreted as a
+        luminance volume.
+    colormap : 2-tuple of str, vispy.color.Colormap
+        The first is the name of the current colormap, and the second value is
+        the colormap. Colormaps are used for luminance volumes, if the volume is
+        multichannel the colormap is ignored.
+    colormaps : tuple of str
+        Names of the available colormaps.
+    clim : list (2,) of float
+        Color limits to be used for determining the colormap bounds for
+        luminance volumes. If the volume is multichannel the clim is ignored.
+    clim_range : list (2,) of float
+        Range for the color limits for luminace volumes. If the volume is
+        multichannel the clim_range is ignored.
+
+    Extended Summary
+    ----------
+    _data_view : array (N, M, K), (N, M, K, 3), or (N, M, K, 4)
+        Volume data for the currently viewed slice. Must be 3D volume data, but
+        can be multidimensional for RGB or RGBA volumes if multidimensional is
+        `True`.
+    """
+
+    class_keymap = {}
+    _colormaps = COLORMAPS_3D_DATA
+
+    def __init__(
+        self,
+        volume,
+        *,
+        metadata=None,
+        multichannel=None,
+        colormap='fire',
+        clim=None,
+        clim_range=None,
+        interpolation='nearest',
+        name=None,
+        **kwargs,
+    ):
+
+        visual = VolumeNode(volume)
+        super().__init__(visual, name)
+
+        self.translate = (0, 0, 0)
+        self.scale = (0.009, 0.009, 0.009, 1)
+
+        self.events.add(clim=Event, colormap=Event)
+
+        with self.freeze_refresh():
+            # Set data
+            self._data = volume
+            self.metadata = metadata or {}
+            self.multichannel = multichannel
+
+            # Intitialize volume views and thumbnails with zeros
+            if self.multichannel:
+                self._data_view = np.zeros((1, 1, 1) + (self.shape[-1],))
+            else:
+                self._data_view = np.zeros((1, 1, 1))
+            self._data_thumbnail = self._data_view
+
+            # Set colormap
+            self._colormap_name = ''
+            self.colormap = colormap
+
+            # Set update flags
+            self._need_display_update = False
+            self._need_visual_update = False
+
+            # Re intitialize indices
+            self._indices = (
```

Comment:
This should still have the ndims code as before to handle 4D datasets where we want the last 3 to be as a volume

Reply:
please fix this to enable the `4D` example to work 
```python
            self._indices = (0,) * (self.ndim - 3) + (
                slice(None, None, None),
                slice(None, None, None),
                slice(None, None, None),
            )
```